pkg/apis/agent/v1alpha1/agent_types.go

```diff
@@ -66,6 +66,30 @@ type AgentSpec struct {
 	// Cannot be used along with `daemonSet`.
 	// +kubebuilder:validation:Optional
 	Deployment *DeploymentSpec `json:"deployment,omitempty"`
+
+	// HTTP holds the HTTP layer configuration for the Agent in Fleet mode with Fleet Server enabled.
+	// +kubebuilder:validation:Optional
+	HTTP commonv1.HTTPConfig `json:"http,omitempty"`
+
+	// Mode specifies the source of configuration for the Agent. The configuration can be specified locally through
+	// `config` or `configRef` (`standalone` mode), or come from Fleet during runtime (`fleet` mode).
+	// Defaults to `standalone` mode.
+	// +kubebuilder:validation:Optional
+	Mode AgentMode `json:"mode,omitempty"`
+
+	// FleetServerEnabled determines whether this Agent will launch Fleet Server. Don't set unless `mode` is set to `fleet`.
+	// +kubebuilder:validation:Optional
+	FleetServerEnabled bool `json:"fleetServerEnabled,omitempty"`
+
+	// KibanaRef is a reference to Kibana where Fleet should be set up and this Agent should be enrolled. Don't set
+	// unless `mode` is set to `fleet`.
+	// +kubebuilder:validation:Optional
+	KibanaRef commonv1.ObjectSelector `json:"kibanaRef,omitempty"`
+
+	// FleetServerRef is a reference to Fleet Server that this Agent should connect to to obtain it's configuration.
+	// Don't set unless `mode` is set to `fleet`.
+	// +kubebuilder:validation:Optional
+	FleetServerRef commonv1.ObjectSelector `json:"fleetServerRef,omitempty"`
```

Comment:
To better encapsulate fleet stuff I'm wondering if we could reorganize this into:

```
fleet:
    managed: true
    server: false
    serverRef: <>
    kibanaRef: <>
```

Reply:
I think this is a viable idea, but it would still require to define `elasticsearchRefs` outside of `fleet` field. This feels confusing as it is a required field for Fleet Server. It allows to tie together some of the fields, but users still need to know how it should be used with Fleet Server:
```yaml
elasticsearchRefs:
- name: es
fleet:
    managed: true
    server: true
    serverRef: {}
    kibanaRef: kb
```

and plain Elastic Agent:
```yaml
elasticsearchRefs: {}
fleet:
    managed: true
    server: false
    serverRef: fs
    kibanaRef: kb
```

Overall, I'm not sure if it's an improvement over keeping everything at the same level.